### DISCUSSION TITLE
Does lvim do custom syntax highlighting with/outside of treesitter?

### DISCUSSION THREAD

**Discussion by nocode99** [LunarVim/LunarVim#3772] [General]
I was trying to use lunarvim but currently switching back to my own neovim. I'm curious where any custom syntax highlighting may be for lunarvim? When I edit terraform files, the syntax highlighting is a bit different than my fairly vanilla treesitter/tokyonight setup

**Comment by LostInTheLogs** [comment]
make sure you have the treesitter parser installed, also the default theme is similar to but it's not tokyonight

**Comment by nocode99** [comment]
First off - thanks for your responses. I won't update the discord thread and keep the convo here. I've done a vanilla install of neovim in a docker image installing the bare minimum and still cannot get the syntax highlighting to match what LunarVim has out of the box. You mentioned that 99% of it is the default, where can I find the 1%?

![image](https://user-images.githubusercontent.com/4175239/214383915-d2904d9a-9d1b-4664-96f2-e66bcbabac34.png)

**Reply by LostInTheLogs** [reply]
only these ones are different https://github.com/LunarVim/LunarVim/blob/8ea68830a0212047aae924cc32ef9d7bb3bb270b/lua/lvim/core/autocmds.lua#L113-L120

other things that may change the highlighting are:
  - lsp: `:LspStop` all lsp servers in both lvim and nvim
  - make sure that the colorscheme is the same, the best way to check would be to use lunar(or any that is not tokyonight) in both, as it didn't change for a longer time

**Comment by nocode99** [comment]
Ah yeah - forgot to mention I tried `lunar` as well and got the same result. I'm gonna take a look at the settings you just posted. Thanks!

**Comment by nocode99** [comment]
So it seems LunarVim is being shipped with older version of treesitter parsers. I tried combing through the code to see where this is configured but couldn't find it. Ultimately, the parser for `hcl` is definitely using an old version and they've now added `terraform` parser which inherits from `hcl`.

I'm opening a PR with `nvim-treesitter` to update the highlighting, but that seems to be the difference. I looked at where `nvim-treesitter` was installed for nvim and lunarvim and that's where I saw the discrepancy in the `queries/hcl/highlights.scm` file

**Comment by LostInTheLogs** [comment]
we pin the versions of our plugins, 1.2 has quite old ones
